### ISSUE TITLE
Playlist feature glitches when adding audios into the playlist on ios

### ISSUE THREAD

**Issue by rootux** [ryanheise/just_audio#1101]
**Which API doesn't behave as documented, and how does it misbehave?**
Only on iOS (Works on Android)
When trying to use the playlist feature.
When already playing some type of audio. upon calling `await _playlist.add(audio)` OR `await _playlist.addAll(audios)`
The player jumps directly to the next song instead of queuing it in the playlist.

**Minimal reproduction project**
Provide a link here using one of two options:
1. Forked the repository - link here - https://github.com/rootux/just_audio/blob/minor/just_audio/example/lib/example_playlist.dart
2. "The example"

**To Reproduce (i.e. user steps, not code)**
```
1. Start the playlist example - https://github.com/rootux/just_audio/blob/minor/just_audio/example/lib/example_playlist.dart
2. Run on IOS (Android works)
3. Click on Play
4. Click rapidly on + to add new items to the playlist
5. observe that the playlist "jumps" and doesn't play fluently

```

**Expected behavior**
Expecting the playlist to finish playing current song and then jump to the next one without any skips.

**Smartphone (please complete the following information):**
 - Device: iPad 2021, Iphone Emulator iphone 14 Pro

**Flutter SDK version**
```
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.7, on macOS 13.4 22F66 darwin-arm64, locale en-IL)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.2)
[✓] VS Code (version 1.79.1)
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!

```

**Additional context**
This happens only for the assets audio source that's why I kept the nature sounds.
Same happens when using local files
```

**Comment by gungun974** [comment]
have you found any solution about this issue ?
I want to create my own Music player for Linux, android and IOS and with the strange way I decided to build my app.
I need to dynamically add song in the `ConcatenatingAudioSource` but since on IOS I can't add the next sound. I don't know what I can do as workaround.

**Comment by rootux** [comment]
Yes. Ive implemented my own playlist feature and stopped using the player for playlists on ios (or for all platforms)

**Comment by gungun974** [comment]
> Yes. Ive implemented my own playlist feature and stopped using the player for playlists on ios (or for all platforms)

Interesting, I am personally very tempted to look for the problem myself in the Objective C source code since it's my favorite way to contribute.
But by curiosity. Have you implement your own playlist feature still with `just_audio` and if yes. Have you manage to make the transition between tracks smooth ?

**Comment by rootux** [comment]
Thats the correct way to solve it.
I tried looking into the swift code and could not understand why it doesnt
work.
So yes. I wrote it over just audio in the flutter side and its smooth.
I listens for audio finish callback and check if more audio in playlist and
if so I push the next one and play it

**Comment by gungun974** [comment]
> **Which API doesn't behave as documented, and how does it misbehave?** Only on iOS (Works on Android) When trying to use the playlist feature. When already playing some type of audio. upon calling `await _playlist.add(audio)` OR `await _playlist.addAll(audios)` The player jumps directly to the next song instead of queuing it in the playlist.
> 
> **Minimal reproduction project** Provide a link here using one of two options:
> 
>     1. Forked the repository - link here - https://github.com/rootux/just_audio/blob/minor/just_audio/example/lib/example_playlist.dart
> 
>     2. "The example"
> 
> 
> **To Reproduce (i.e. user steps, not code)** Steps to reproduce the behavior:
> 
> ```
> 1. Start the playlist example - https://github.com/rootux/just_audio/blob/minor/just_audio/example/lib/example_playlist.dart
> 2. Run on IOS (Android works)
> 3. Click on Play
> 4. Click rapidly on + to add new items to the playlist
> 5. observe that the playlist "jumps" and doesn't play fluently
> ```
> 
> **Expected behavior** Expecting the playlist to finish playing current song and then jump to the next one without any skips.
> 
> **Smartphone (please complete the following information):**
> 
>     * Device: iPad 2021, Iphone Emulator iphone 14 Pro
> 
> 
> **Flutter SDK version**
> 
> ```
> flutter doctor
> Doctor summary (to see all details, run flutter doctor -v):
> [✓] Flutter (Channel stable, 3.13.7, on macOS 13.4 22F66 darwin-arm64, locale en-IL)
> [✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
> [✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
> [✓] Chrome - develop for the web
> [✓] Android Studio (version 2022.2)
> [✓] IntelliJ IDEA Ultimate Edition (version 2023.1.2)
> [✓] VS Code (version 1.79.1)
> [✓] Connected device (2 available)
> [✓] Network resources
> 
> • No issues found!
> ```
> 
> **Additional context** This happens only for the assets audio source that's why I kept the nature sounds. Same happens when using local files

Well it's very strange for me.

I try to replicate your issue with your example and I got a perfectly normal player.

I reemplement on my end my own testing bench and I don't got the strange track jump behavior.

But something in my main code in production make the audio player on IOS only behave like you explain.

I need more investigation (which is on my way) but it's more a strange and obscure bug.

**Comment by gungun974** [comment]
I finally manage to find something but in my code it's when I add a network Audio file where the player glitches to the next audios.

But when I use HLS audios, I don't have any issue. So I solve my issue since my audio server support HLS streaming but it's really weird I got issue when I try to load file like flac.
